week_2/pet.rb

```diff
@@ -90,26 +95,40 @@ def food_cost(days)
   # snake = Pet.new(name: 'python', animal_type_id: 4, food_consumed_per_day: 0.3)
   # Pet.cost_to_feed([cat, dog, fish, snake], 6) will return 6180.0
   def self.cost_to_feed(pets, days)
-    raise NotImplementedError # TODO
+    c=0
+   
+  pets.each do |an|
+    c=c+(FOOD_COST_PER_KG[an.animal_type_id]*days*an.food_consumed_per_day)
+  end
+    return c
   end
 
   # This function takes an array of pets as input
   # and returns a hash with pets name grouped by their animal type
   #
   # Eg - 
-  # cat = Pet.new(name: 'cat', animal_type_id: 6, food_consumed_per_day: 0.4)
+ 
+  #  cat = Pet.new(name: 'cat', animal_type_id: 6, food_consumed_per_day: 0.4)
   # dog = Pet.new(name: 'dog', animal_type_id: 6, food_consumed_per_day: 0.7)
   # fish = Pet.new(name: 'clownfish', animal_type_id: 2, food_consumed_per_day: 0.1)
   # snake = Pet.new(name: 'python', animal_type_id: 4, food_consumed_per_day: 0.3)
-  # Pet.group_by_animal_type([cat, dog, fish, snake]) will return the follwing hash
+  #  Pet.group_by_animal_type([cat, dog, fish, snake]) will return the follwing hash
   # {
   #   6 => ['cat', 'dog'],
   #   2 => ['clownfish'],
   #   4 => ['python']
   # }
-  #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
-  end
+ans={}
+for pet in pets do
+  if(ans.keys.include?pet.animal_type_id)
```

Comment:
```suggestion
  if !ans[pet.animal_type_id]
```
`nil` is returned if the key doesn't exist